docs/package-migration-process-guide.md

```diff
@@ -0,0 +1,241 @@
+
+### **[PR#3]** 5. Review the `metamask-module-template`, and add any missing files or elements (e.g. LICENSE)
+
+- [Example PR](https://github.com/MetaMask/eth-json-rpc-provider/pull/24)
+
+### **[PR#4]** 6. Rename the migration target package so that it is prepended by the `@metamask/` namespace (skip if not applicable)
+
+- Modify the "name" field in `package.json`.
+- Update the title of the README.md.
+- Add a CHANGELOG entry for the rename.
+
+### **[PR#5]** 7. Create a new release of the migration target from the source repo
+
+- All subsequent releases of the migration target will be made from the core monorepo.
+- [Example PR](https://github.com/MetaMask/eth-json-rpc-provider/pull/29)
+
+## Phase B: **Staging** from the core monorepo's `merged-packages/` directory
+
+### **[PR#6]** 1. Migrate the source repo's git history into the `merged-packages/` temporary directory in core
+
+#### Steps
+
+1. [Install `git-filter-repo`](https://github.com/newren/git-filter-repo/blob/main/INSTALL.md). This tool is like Git's `filter-branch` command, but much easier to use and much less error-prone.
+2. Navigate to a temporary directory: `cd /tmp`
+3. Clone the repo for the library you want to migrate: `git clone https://github.com/MetaMask/<repo-name>` (e.g. `<repo-name>` = `utils`). Do **NOT** use an existing clone.
+4. `cd` into the newly cloned repo.
+5. Run `git filter-repo --to-subdirectory-filter merged-packages/<package-name>` (e.g. `<package-name>` = `utils`). This will rewrite all history to move all files to `merged-packages/<package-name>`. (This is why we're making a fresh clone in a temporary directory — this action is irreversible.)
+6. `cd` to wherever you keep the `core` repo on your machine.
+7. Add the library as a remote: `git remote add <package-name> /tmp/<package-name>`.
+8. Fetch history: `git fetch <package-name> --no-tags`
+9. Make a new branch: `git checkout -b migrate-<package-name>`
+10. Merge the library into the monorepo: `git merge --allow-unrelated-histories <package-name>/<primary-branch>` (e.g. `<primary-branch>` = `main`)
+11. Open a pull request in the core repo that reflects the above changes.
+
+#### Caution
+
+- DO NOT rebase the `migrate-<package-name>` branch, as this will disrupt the git history.
+- Ensure that superfluous merge commits to the main branch don't pollute the migrated git history.
+  - Coordinate with the team to minimize the time that this PR stays open.
+  - If necessary, replace the PR branch with a cleaned-up commit history by rerunning the git history migration steps before merging the PR.
+  - For further context on this: https://github.com/MetaMask/core/pull/1804#issuecomment-1771445829
+- Merge PR **without squashing** to preserve the migrated git commit history.
+  - Contact a [**maintainer**](https://github.com/orgs/MetaMask/teams/engineering?query=role%3Amaintainer) to temporarily enable merge commits into main.
+- [Example PR](https://github.com/MetaMask/core/pull/1872)
```

Comment:
Nit: Worth using an emoji to make this pop out?

``` suggestion
#### ⚠️ Caution

- DO NOT rebase the `migrate-<package-name>` branch, as this will disrupt the git history.
- Ensure that superfluous merge commits to the main branch don't pollute the migrated git history.
  - Coordinate with the team to minimize the time that this PR stays open.
  - If necessary, replace the PR branch with a cleaned-up commit history by rerunning the git history migration steps before merging the PR.
  - For further context on this: https://github.com/MetaMask/core/pull/1804#issuecomment-1771445829
- Merge PR **without squashing** to preserve the migrated git commit history.
  - Contact a [**maintainer**](https://github.com/orgs/MetaMask/teams/engineering?query=role%3Amaintainer) to temporarily enable merge commits into main.
- [Example PR](https://github.com/MetaMask/core/pull/1872)
```

or, use a WARNING block:

``` suggestion
> [!WARNING]
> - DO NOT rebase the `migrate-<package-name>` branch, as this will disrupt the git history.
> - Ensure that superfluous merge commits to the main branch don't pollute the migrated git history.
>   - Coordinate with the team to minimize the time that this PR stays open.
>   - If necessary, replace the PR branch with a cleaned-up commit history by rerunning the git history migration steps before merging the PR.
>   - For further context on this: https://github.com/MetaMask/core/pull/1804#issuecomment-1771445829
> - Merge PR **without squashing** to preserve the migrated git commit history.
>   - Contact a [**maintainer**](https://github.com/orgs/MetaMask/teams/engineering?query=role%3Amaintainer) to temporarily enable merge commits into main.
> - [Example PR](https://github.com/MetaMask/core/pull/1872)
```

Whichever looks better to you.

Reply:
Went with the second option here: https://github.com/MetaMask/core/pull/2071/commits/a25df15255039d459947cd1133201589c8533a5d